aea/cli/push.py

```diff
@@ -125,10 +123,11 @@ def _check_package_public_id(source_path, item_type, item_id) -> None:
     item_author = config.get("author", "")
     item_name = config.get("name", "")
     item_version = config.get("version", "")
+    actual_item_id = PublicId(item_author, item_name, item_version)
     if (
-        item_id.name != item_name
-        or item_id.author != item_author
-        or item_id.version != item_version
+        actual_item_id.same_prefix(item_id)
+        and not item_id.latest
+        and item_id.version != item_version
```

Comment:
This looks wrong..(we want to raise if the package specified in command and package config *don't* match)